src/custom/state/orders/updater.ts

```diff
@@ -104,54 +130,76 @@ export function EventUpdater(): null {
       console.log('EventUpdater::getLogs', {
         fromBlock: lastCheckedBlock + 1,
         toBlock: lastBlockNumber,
-        // address: '0x',
-        topics: TransferEventTopics
+        // address: '0x', // TODO: get address from networks.json or somewhere else
+        topics: eventTopics
       })
 
       const logs = await getLogsRetry({
         fromBlock: lastCheckedBlock + 1,
         toBlock: lastBlockNumber,
-        // address: '0x',
-        topics: TransferEventTopics
+        // address: '0x', // TODO: get address from networks.json or somewhere else
+        topics: eventTopics
       })
 
-      // console.log('logs', logs)
-      logs.forEach(log => {
-        try {
-          // const { from, to, amount } = decodeTransferEvent(log)
-          // const { from, to, amount, id } = decodeTransferEvent(log)
-
-          console.log('EventUpdater::Detected transfer of token in block', log.blockNumber)
-          // console.log('EventUpdater::Detected transfer of token since last time', log.address, { from, to, amount })
-          // dispatch(fulfillOrder({ chainId, id, fulfillmentTime: from log,if: from log }))
-        } catch (e) {}
+      batch(() => {
+        logs.forEach(log => {
+          try {
+            // const { from, to, amount } = decodeTransferEvent(log)
+            const { orderUid: id } = decodeTradeEvent(log)
+
+            console.log(`EventUpdater::Detected Trade event for order ${id} of token in block`, log.blockNumber)
+            dispatch(
+              fulfillOrder({
+                chainId,
+                id,
+                fulfillmentTime: new Date().toISOString() // event only has blockNumber
```

Comment:
but you can get the block using the library, right?

Reply:
Yes, and I will do it, with proper caching,so that we don't spam blockchain with eth_getBlock requests
In a separate PR